ghcide.cabal

```diff
@@ -221,4 +222,16 @@ test-suite ghcide-tests
         Development.IDE.Test
         Development.IDE.Test.Runfiles
     default-extensions:
+        BangPatterns
```

Comment:
This change looks unrelated to this PR and you barely make use of it (judging from the fact that you didn’t remove any extensions and you only added one module so you could just add all extensions there.

Reply:
I made it because I wasted a couple of minutes rewriting my test code to work around hlint errors. 
```
  # Extensions enabled by `bazel` and `da-ghci` by default.  We ban them here
  # to avoid useless pragmas piling up on the top of files.
  - {name: BangPatterns, within: []}
  - {name: DeriveDataTypeable, within: []}
  - {name: DeriveFoldable, within: []}
  - {name: DeriveFunctor, within: []}
  - {name: DeriveGeneric, within: []}
  - {name: DeriveTraversable, within: []}
  - {name: FlexibleContexts, within: []}
  - {name: GeneralizedNewtypeDeriving, within: []}
  - {name: LambdaCase, within: []}
  - {name: NamedFieldPuns, within: []}
  - {name: OverloadedStrings, within: []}
  - {name: PackageImports, within: []}
  - {name: RecordWildCards, within: []}
  - {name: ScopedTypeVariables, within: []}
  - {name: StandaloneDeriving, within: []}
  - {name: TupleSections, within: []}
  - {name: TypeApplications, within: []}
  - {name: ViewPatterns, within: []}
```
It's unrelated to this PR indeed, but also too small to justify a PR on its own.